src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CaseManagementCategoryAppender.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AppealType;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.DynamicList;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.Value;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class CaseManagementCategoryAppender implements PreSubmitCallbackHandler<AsylumCase> {
+
+    public boolean canHandle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && Arrays.asList(
+                Event.START_APPEAL,
+                Event.EDIT_APPEAL,
+                Event.EDIT_APPEAL_AFTER_SUBMIT).contains(callback.getEvent());
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+            PreSubmitCallbackStage callbackStage,
+            Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        Optional<AppealType> appealType
```

Comment:
Is an appealType a `must` in a case?  In which case if it's not present I would suggest throwing an illegal state exception.  @makowalski what do you think?

Reply:
We also need to think about historic case data that will not have the new field present.  We have a pattern whereby we can fill these in on callbacks such as here: https://github.com/hmcts/ia-case-api/blob/master/src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AddBaseLocationFromHearingCentreForOldCasesFixHandler.java

The only gap then will be cases where no CCD Events have run successfully since we implement the above

Reply:
@raghera @mkayad 
appelType is mandatory field on startAppel event (first event in the appeal lifecycle). As Ravi suggested we should use Illegal State Exception to handle missing appealType field. We are doing it in several places in code already.
```
  {
    "LiveFrom": "01/01/2018",
    "CaseTypeID": "Asylum",
    "CaseEventID": "startAppeal",
    "CaseFieldID": "appealType",
    "CaseEventFieldLabel": "Type of appeal",
    "PageFieldDisplayOrder": 2,
    "DisplayContext": "MANDATORY",
    "PageID": "appealType",
    "PageDisplayOrder": 18,
    "ShowSummaryChangeOption": "Y"
  },
```

Reply:
Done